catalog-scripts/register_discovered_locations_bitbucket.py

```diff
@@ -0,0 +1,84 @@
+import requests
+import re
+from requests.auth import HTTPBasicAuth
+from requests.packages.urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
+
+username = "" # Your Bitbucket username
+workspace = "" # Your Bitbucket workspace
+app_password = "" # Your Bitbucket app_password
+repository = "" # Your Bitbucket repo
+branch = "" # Your Bitbucket branch
+prefix_path = "mock_rserver_root/configs/services/" # Your prefix path
+catalog_path = r"mock_rserver_root/configs/services/.*?-service" # Your suffix path
+
+account = "" 
+# Replace with your x-api-key. Refer https://developer.harness.io/docs/platform/automation/api/api-quickstart/#create-a-harness-api-key-and-token to generate one
+x_api_key = ""
+
+api_url = f"https://idp.harness.io/{account}/idp/api/catalog/locations"
+
+directories = []
+dir_url = f"https://api.bitbucket.org/2.0/repositories/{workspace}/{repository}/src/{branch}/{prefix_path}"
+response = requests.get(dir_url, auth=HTTPBasicAuth(username, app_password))
+if response.status_code != 200:
+    print(f"Failed to fetch the latest commit. Status code: {response.status_code}")
+    exit()
+
+def extract_directories(response):
```

Comment:
Can we move this method to the top (after the variable declarations) OR to the bottom? 
Right now it's kind of in between the code flow which affects readability